File: TableViewObjC/TableViewObjC/ViewController.m
//
//  ViewController.m
//  TableViewObjC
//
//  Created by Osman SÖYLEMEZ on 01/03/2018.
//  Copyright © 2018 Osman SÖYLEMEZ. All rights reserved.
//

#import "ViewController.h"

static NSString *tableViewCellIdentifier = @"TableViewCell";

@interface ViewController () <UITableViewDelegate, UITableViewDataSource>

@property (weak, nonatomic) IBOutlet UITableView *tableView;

@property (nonatomic, strong) NSArray *dataSource;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.dataSource = @[
                        @"Pazartesi",
                        @"Salı",
                        @"Çarşamba",
                        @"Perşembe",
                        @"Cuma",
                        @"Cumartesi",
                        @"Pazar"
                        ];
}

- (void)showAlertWithMessage:(NSString *)message
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Uyarı"
                                                                             message:message
                                                                      preferredStyle:UIAlertControllerStyleAlert];

    UIAlertAction *ok = [UIAlertAction actionWithTitle:NSLocalizedString(@"Tamam", nil)
                                                 style:UIAlertActionStyleDefault
                                               handler:nil];
    [alertController addAction:ok];
    
    [self presentViewController:alertController animated:YES completion:nil];
}

#pragma mark UITableView

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.dataSource.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:tableViewCellIdentifier];
    
    if (cell == nil)
    {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:tableViewCellIdentifier];
    }
    
    cell.textLabel.text = [self.dataSource objectAtIndex:indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSString *item = [self.dataSource objectAtIndex:indexPath.row];
    [self showAlertWithMessage:item];
}

@end

File: TableViewSwift/TableViewSwift/ViewController.swift
//
//  ViewController.swift
//  TableViewSwift
//
//  Created by Osman SÖYLEMEZ on 01/03/2018.
//  Copyright © 2018 Osman SÖYLEMEZ. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    @IBOutlet weak var tableView: UITableView!
    
    var dataSource: [String]?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.dataSource = [
                            "Pazartesi",
                            "Salı",
                            "Çarşamba",
                            "Perşembe",
                            "Cuma",
                            "Cumartesi",
                            "Pazar",
                            ]
    }
    
    func showAlert(message: String) {
        
        let alertController = UIAlertController(title: "Uyarı",
                                                message: message,
                                                preferredStyle: .alert)
        
        let alertAction = UIAlertAction.init(title: NSLocalizedString("Tamam", comment: "comment"),
                                             style: .default,
                                             handler: nil)
        alertController.addAction(alertAction)
        self.present(alertController, animated: true, completion: nil)
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.dataSource!.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let tableViewCellIdentifier = "TableViewCell"
        var cell = tableView.dequeueReusableCell(withIdentifier: tableViewCellIdentifier)
        
        if (cell == nil)
        {
            cell = UITableViewCell.init(style: .default, reuseIdentifier: tableViewCellIdentifier)
        }
        
        cell?.textLabel?.text = self.dataSource?[indexPath.row]
        
        return cell!
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        self.showAlert(message: self.dataSource![indexPath.row])
    }
}

